gravatar/src/main/java/com/gravatar/di/container/GravatarSdkContainer.kt

```diff
@@ -31,12 +31,12 @@ internal class GravatarSdkContainer private constructor() {
 
     fun getGravatarV3Service(okHttpClient: OkHttpClient? = null, oauthToken: String? = null): GravatarApi {
         return getRetrofitApiV3Builder().apply {
-            client(okHttpClient ?: buildOkHttpClient(oauthToken))
+            client(okHttpClient.buildOkHttpClient(oauthToken))
         }.addConverterFactory(GsonConverterFactory.create(gson))
             .build().create(GravatarApi::class.java)
     }
 
-    private fun buildOkHttpClient(oauthToken: String?) = OkHttpClient()
+    private fun OkHttpClient?.buildOkHttpClient(oauthToken: String?) = (this ?: OkHttpClient())
```

Comment:
We'll need to inform developers that we are overriding the `OkHttpClient` to avoid errors. I don't think several developers will provide a custom client, but just in case.

Reply:
I added an explanation in the docs.